lib/tasks/data-key.rake

```diff
@@ -8,4 +8,63 @@ namespace :"data-key" do
     key = Base64.strict_encode64(Slosilo::Symmetric.new.random_key)
     print key
   end
+
+  desc "Validates the value of ENV['CONJUR_DATA_KEY']"
+  # This task validates the master data key that is stored in `ENV['CONJUR_DATA_KEY']`
+  # The task runs in two modes:
+  # First mode:
+  # ==========
+  #   Once the database is created, a role + resource + secret will be created using the master key.
+  #
+  # Second mode:
+  # ===========
+  #   Each time the server restarts this task will attempt to retrieve the
+  #   secret. if the provided key does not match the original Conjur data key, the secret retrieval
+  #   will fail and the task will exit with error.
+  #
+  # Side Note
+  # =========
+  #  What's the 'environment' task in Rake?
+  #  the :validate => :environment means "do environment before validate".
+  #  You can get access to your models, and in fact, your whole environment
+  #  by making tasks dependent on the environment task.
+  task :validate => [:environment] do
+    data_key = ENV['CONJUR_DATA_KEY']
+
+    if data_key.nil? || data_key.empty?
+      raise Errors::System::EmptyConjurDataKey
+      exit 1
+    end
+
+    # try get the Conjur data key validator resource
+    resource_id = "!:!:data-key-validator-resource"
+    if Resource[resource_id]
+      # Server restart - verify we can decrypt the secret
+      begin
+        # Attempt to retrieve the secret encrypt with the original data key.
+        # A failure to decrypt the value will indicate the key is invalid.
+        Resource[resource_id].secret.value
+      rescue => e
+        if e.inspect == 'OpenSSL::Cipher::CipherError'
```

Comment:
Instead of using `inspect` to do an implicit string conversion and matching on strings, just rescue the specific error `OpenSSL::Cipher::CipherError`, and then add a 2nd rescue for everything else.  Then you can delete the `if... else` currently inside the `rescue`.